File: tut-vaadin-simple-javascript-component-integration/src/main/java/com/github/galcyurio/ui/MainUI.java
package com.github.galcyurio.ui;

import com.github.galcyurio.client.MyComponent;
import com.vaadin.server.VaadinRequest;
import com.vaadin.spring.annotation.SpringUI;
import com.vaadin.ui.UI;
import com.vaadin.ui.VerticalLayout;

/**
 * @author galcyurio
 */
@SpringUI
public class MainUI extends UI {

    @Override
    protected void init(VaadinRequest request) {
        setContent(new VerticalLayout() {{
            addComponents(new MyComponent() {{
                setValue("Hello, MyComponent!");
            }});
        }});
    }
}

File: tut-vaadin-simple-webjars/src/main/kotlin/com/github/galcyurio/ui/MainUI.kt
package com.github.galcyurio.ui

import com.vaadin.server.VaadinRequest
import com.vaadin.spring.annotation.SpringUI
import com.vaadin.ui.*


/**
 * @author galcyurio
 */
@com.vaadin.annotations.JavaScript(
    "https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js",
    "vaadin://js/test.js")
@SpringUI
class MainUI : UI() {
    val textField = TextField()
    val button = Button("javascript alert")

    override fun init(request: VaadinRequest?) {
        content = VerticalLayout().apply {
            addComponents(
                textField,
                button.apply {
                    addClickListener {
                        JavaScript.getCurrent().execute("""alertText("${textField.value}")""")
                    }
                }
            )
        }
    }
}